app/javascript/components/map/drawing-manager/component.js

```diff
@@ -0,0 +1,58 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+
+// Constants
+import { DRAWING_OPTIONS } from './constants';
+
+// styles
+// import './styles.scss';
+
+class DrawingManagerComponent extends PureComponent {
+  static propTypes = {
+    map: PropTypes.object.isRequired,
+    drawing: PropTypes.bool.isRequired,
+    setDrawing: PropTypes.func.isRequired,
+    setClearing: PropTypes.func.isRequired,
+    setAreaOfInterestArea: PropTypes.func.isRequired
+  }
+
+  componentDidMount() {
+    this.props.map.on('pm:drawstart', (e) => {
+      if (this.layer) {
+        this.layer.remove();
+        this.props.setAreaOfInterestArea({});
+      }
+    });
+
+    this.props.map.on('pm:create', (e) => {
+      this.layer = e.layer;
+      this.props.setAreaOfInterestArea(e.layer.toGeoJSON());
+      this.props.setDrawing(false);
+    });
+  }
+
+  componentDidUpdate(prevProps) {
+    const { drawing: prevDrawing } = prevProps;
+    const { drawing: nextDrawing, clearing } = this.props;
+
+    if (clearing) {
+      this.layer.remove();
```

Comment:
Always check that `this.layer` exists. You could have an error if the update is triggered and there isn't a layer due to any reason